### PR TITLE
added subscription environment variable for SYNC-PERIOD-MIN

### PR DESCRIPTION
**Pull request by redhatHameed**
The operator does not allow users to change environment variables. All changes get reverted. the possible solution is to add into subscription config: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/subscription-config.md#subscription-config

add filed SyncPeriod into DBaaS-platform CR to make it configurable. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: api/v1alpha1/dbaasplatform_types.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ const (
 
 // DBaaSPlatformSpec defines the desired state of DBaaSPlatform
 type DBaaSPlatformSpec struct {
-	Name string `json:"name,omitempty"`
+	SyncPeriod string `json:"syncPeriod,omitempty"`
```
Comment (tchughesiv) [inline api/v1alpha1/dbaasplatform_types.go:44]: would it be a good idea to make this an integer instead? then convert to string in code?

Comment (redhatHameed) [reply api/v1alpha1/dbaasplatform_types.go]: make sense, added validation as well. updated

File: config/crd/bases/dbaas.redhat.com_dbaasplatforms.yaml [hunk 2]
```diff
@@ -36,8 +36,10 @@ spec:
           spec:
             description: DBaaSPlatformSpec defines the desired state of DBaaSPlatform
             properties:
-              name:
-                type: string
+              syncPeriod:
+                maximum: 180
```
Comment (tchughesiv) [inline config/crd/bases/dbaas.redhat.com_dbaasplatforms.yaml:40]: any reason 180 is the max here? just curious

**Review by tchughesiv — CHANGES_REQUESTED** [review]
can we add a couple unit tests that verify no ENV is set when `cr.Spec.SyncPeriod` is empty? and another that verifies one IS set? also, what happens if an empty `SYNC-PERIOD-MIN` var is set? does that mean no refresh?

File: api/v1alpha1/dbaasplatform_types.go [hunk 3]
```diff
@@ -41,7 +41,9 @@ const (
 
 // DBaaSPlatformSpec defines the desired state of DBaaSPlatform
 type DBaaSPlatformSpec struct {
-	Name string `json:"name,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=180
+	SyncPeriod int `json:"syncPeriod,omitempty"`
```
Comment (tchughesiv) [inline api/v1alpha1/dbaasplatform_types.go:46]: can we add a description here?

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: config/crd/bases/dbaas.redhat.com_dbaasplatforms.yaml (hunk 2 above)
Comment (jianrongzhang89) [reply config/crd/bases/dbaas.redhat.com_dbaasplatforms.yaml:40]: @redhatHameed you can set default: 180 instead of maximum.

Comment (redhatHameed) [reply config/crd/bases/dbaas.redhat.com_dbaasplatforms.yaml]: @jianrongzhang89 Thanks default already set by provider operator.

@tchughesiv added 24h max value

File: api/v1alpha1/dbaasplatform_types.go (hunk 3 above)
Comment (redhatHameed) [reply api/v1alpha1/dbaasplatform_types.go]: added

**Comment by redhatHameed** [comment]
> can we add a couple unit tests that verify no ENV is set when `cr.Spec.SyncPeriod` is empty? and another that verifies one IS set? also, what happens if an empty `SYNC-PERIOD-MIN` var is set? does that mean no refresh?

added unit test.

added if check if empty/nil cr.Spec.SyncPeriod don't set environment variable

**Review by tchughesiv — APPROVED** [review]
Lgtm

File: controllers/dbaasplatform_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+				Labels:    map[string]string{"managed-by": "dbaas-operator"},
+			},
+
+			Spec: dbaasv1alpha1.DBaaSPlatformSpec{
+
+				SyncPeriod: nil,
+			},
+		}
+		BeforeEach(assertResourceCreation(cr))
+		AfterEach(assertResourceDeletion(cr))
```
Comment (tchughesiv) [inline controllers/dbaasplatform_controller_test.go:45]: Sorry, Can we check the resulting subscription here? See if env var is set or not

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: controllers/dbaasplatform_controller_test.go (hunk 4 above)
Comment (redhatHameed) [reply controllers/dbaasplatform_controller_test.go:45]: This required a complete platform installations unit test as currently not added. I have given try but there are some other issues faced, also we have dependencies on platform installation on openshift real environment and installation never completed see logs:
https://gist.github.com/redhatHameed/2ca684b6488b77193ab77b0591c933fb

**Review by tchughesiv — APPROVED** [review]
lgtm

**Pull request merged**